### ISSUE TITLE
Add jsdoc examples to all of your components

### ISSUE THREAD

**Issue by tnrich** [palantir/blueprint#4162]
#### Feature request
Hey @adidahiya and others. Thanks again for blueprintjs, it is great! I love the meticulous typings. One thing I think that could make this package even better would be if it exported a couple real-life usage examples in the .d.ts files. 

#### Examples

Right now for example, when I go to use the <HTMLSelect/> component, this is all the hover displays: 

![image](https://user-images.githubusercontent.com/2730609/83304513-d734c880-a1b3-11ea-9c28-176d75f7b0de.png)

However, if I add an example in a jsdoc comment above the HTMLSelect definition like so:

![image](https://user-images.githubusercontent.com/2730609/83304634-0c411b00-a1b4-11ea-926b-646fd26a9d89.png)

Then we get a nice example when using the element: 
![image](https://user-images.githubusercontent.com/2730609/83304657-1531ec80-a1b4-11ea-8421-82209ed7df98.png)

I think this would make an already easy development flow even better. Thanks!

**Comment by adidahiya** [comment]
That's a nice idea, but I think in practice those code examples would be hard to maintain because they would not be parsed as TS syntax and therefore not verified as valid code in the build. They would get out of date, one more thing to remember to maintain when updating props...

I think a more maintainable / realistic alternative would be adding JSDoc links to the component documentation pages...?

**Comment by tnrich** [comment]
@adidahiya I agree that they would require more upkeep but I still think they would be worth it. That said as a first step having links to the examples would be awesome!
